src/commons/withScrollEnabler.tsx

```diff
@@ -1,74 +1,67 @@
+  scrollEnabled: boolean;
+};
+
+type SupportedViews = FlatListProps<any> | ScrollViewProps;
+
+export type WithScrollEnablerProps = SupportedViews & {
+  scrollEnablerProps: ScrollEnablerProps;
+  ref?: any;
+};
+type PropTypes = ForwardRefInjectedProps & SupportedViews;
+function withScrollEnabler<PROPS extends SupportedViews>(
+  WrappedComponent: React.ComponentType<WithScrollEnablerProps>
+): React.ComponentType<PROPS> {
+  const ScrollEnabler = (props: PropTypes) => {
     const [scrollEnabled, setScrollEnabled] = useState(true);
     const [contentSize, setContentSize] = useState(0);
     const [layoutSize, setLayoutSize] = useState(0);
 
-    const scrollViewRef = useRef<WithScrollEnablerProps>(null);
-
-    useEffect(() => {
-      const {forwardedRef} = props;
-      if (_.isFunction(forwardedRef)) {
-        forwardedRef(scrollViewRef);
-      } else if (_.isObject(forwardedRef)) {
-        forwardedRef.current = scrollViewRef;
-      }
-    }, [scrollViewRef]);
-
     useEffect(() => {
       const isScrollEnabled = contentSize > layoutSize;
       if (isScrollEnabled !== scrollEnabled) {
         setScrollEnabled(isScrollEnabled);
-        if (Constants.isAndroid && scrollViewRef.current && scrollViewRef.current) {
-          if (_.isFunction(scrollViewRef.current.scrollTo)) {
-            // ListView
-            scrollViewRef.current.scrollTo({x: 0, y: 0, animated: false});
-          } else if (_.isFunction(scrollViewRef.current.scrollToIndex)) {
-            // FlatList
-            scrollViewRef.current.scrollToIndex({index: 0, animated: false});
-          }
-        }
       }
     }, [contentSize, layoutSize]);
 
-    const onContentSizeChange = useCallback((contentWidth: number, contentHeight: number) => {
-      const size = props.horizontal ? contentWidth : contentHeight;
-      if (size !== contentSize) {
-        setContentSize(size);
-      }
-      _.invoke(props, 'onContentSizeChange', contentWidth, contentHeight);
-    },
-    [props]);
+    const onContentSizeChange = useCallback(
+      (contentWidth: number, contentHeight: number) => {
+        const size = props.horizontal ? contentWidth : contentHeight;
+        if (size !== contentSize) {
+          setContentSize(size);
+        }
+      },
+      [props]
+    );
 
-    const onLayout = useCallback((event) => {
-      const {
-        nativeEvent: {
-          layout: {width, height}
+    const onLayout = useCallback(
+      (event) => {
+        const {
+          nativeEvent: {
+            layout: {width, height}
+          }
+        } = event;
+        const size = props.horizontal ? width : height;
+        if (size !== layoutSize) {
+          setLayoutSize(size);
         }
-      } = event;
-      const size = props.horizontal ? width : height;
-      if (size !== layoutSize) {
-        setLayoutSize(size);
-      }
-      _.invoke(props, 'onLayout', event);
-    },
-    [props]);
+      },
+      [props]
```

Comment:
same here? 
why do u pass all the props?

Reply:
Fixed and removed - does not need to use state here